### PR TITLE
oVirt Provisioning Support

### PR DESCRIPTION
**Pull request by abutcher**

### CODE REVIEW

File: pkg/apis/hive/v1/ovirt/platform.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Platform stores all the global configuration that all
+// machinesets use.
+type Platform struct {
+	// The target cluster under which all VMs will run
+	ClusterID string `json:"ovirt_cluster_id"`
```
[the remark this answers is not in the document]
Comment (abutcher) [reply pkg/apis/hive/v1/ovirt/platform.go:10]: These match what the installer uses. I think we should keep them?

File: contrib/pkg/utils/ovirt/ovirt.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				return nil, err
+			}
+			if os.IsNotExist(err) {
+				continue
+			}
+			credsFile = filePath
+			break
+		}
+	}
+	// log.WithField("credsFile", credsFile).Info("Loading oVirt creds")
```
Comment (twiest) [inline contrib/pkg/utils/ovirt/ovirt.go:30]: Did you mean to leave this comment in?

File: pkg/clusterresource/builder.go [hunk 3]
```diff
@@ -156,6 +159,9 @@ func (o *Builder) Build() ([]runtime.Object, error) {
 	var allObjects []runtime.Object
 	allObjects = append(allObjects, o.generateClusterDeployment())
 	allObjects = append(allObjects, o.generateMachinePool())
+	if !o.SkipMachinePoolGeneration {
```
Comment (twiest) [inline pkg/clusterresource/builder.go:162]: Hey, I hit you up on slack about the machinepool missing stuff in `platform`. 

Is this why the `platform` part of the machinepool wasn't filled out?

Here's what it looked like:
```yaml
apiVersion: v1
items:
- apiVersion: hive.openshift.io/v1
  kind: MachinePool
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"hive.openshift.io/v1","kind":"MachinePool","metadata":{"annotations":{},"name":"abutcher2-worker","namespace":"default"},"spec":{"clusterDeploymentRef":{"name":"abutcher2"},"name":"worker","platform":{"ovirt":{}},"replicas":3},"status":{}}
    creationTimestamp: "2020-06-29T14:37:15Z"
    finalizers:
    - hive.openshift.io/remotemachineset
    generation: 1
    name: abutcher2-worker
    namespace: default
    resourceVersion: "29035149"
    selfLink: /apis/hive.openshift.io/v1/namespaces/default/machinepools/abutcher2-worker
    uid: 98cd33d3-66e4-4220-a354-700e9d46ab50
  spec:
    clusterDeploymentRef:
      name: abutcher2
    name: worker
    platform:
      ovirt: {}
    replicas: 3
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

File: pkg/clusterresource/ovirt.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+	installertypes "github.com/openshift/installer/pkg/types"
+	installerovirt "github.com/openshift/installer/pkg/types/ovirt"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	hivev1ovirt "github.com/openshift/hive/pkg/apis/hive/v1/ovirt"
+	"github.com/openshift/hive/pkg/constants"
+)
+
+var _ CloudBuilder = (*OpenStackCloudBuilder)(nil)
```
Comment (twiest) [inline pkg/clusterresource/ovirt.go:17]: Should this be `OvirtCloudBuilder`, not  `OpenStackCloudBuilder`?

File: contrib/pkg/utils/ovirt/ovirt.go (hunk 2 above)
Comment (abutcher) [reply contrib/pkg/utils/ovirt/ovirt.go:30]: I'll remove this. There is a similar log like this for another platform but it was messing up `-o yaml`.

File: config/crds/hive.openshift.io_clusterdeployments.yaml [hunk 5]
```diff
@@ -423,6 +423,48 @@ spec:
...
+                    oVirt
+                  properties:
+                    certificatesSecretRef:
+                      description: CertificatesSecretRef refers to a secret that contains
+                        the oVirt CA certificates necessary for communicating with
+                        oVirt.
+                      properties:
+                        name:
+                          description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                            TODO: Add other useful fields. apiVersion, kind, uid?'
```
[the remark this answers is not in the document]
Comment (abutcher) [reply config/crds/hive.openshift.io_clusterdeployments.yaml:437]: The description is generated.

**Comment by twiest** [comment]
These changes all look good to me. Not adding tag since it's still WIP.

File: pkg/clusterresource/builder.go (hunk 3 above)
Comment (abutcher) [reply pkg/clusterresource/builder.go:162]: I think your PR should remove this.

Comment (twiest) [reply pkg/clusterresource/builder.go:162]: Ok, will do.

Just wondering, though, the line above the if statement is exactly the same as the line inside the if statement, so I think that means that you're always generating the `machinepool`.

**Pull request merged**